test/cmd/diff.sh

```diff
@@ -44,6 +44,14 @@ run_kubectl_diff_tests() {
     output_message=$(kubectl diff -f hack/testdata/pod-changed.yaml || test $? -eq 1)
     kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.0'
 
+    # Ensure diff only dry-runs and doesn't persist change
```

Comment:
Can we make sure that both server-side and client-side dry-run are working properly? They both do things a different way so there is a possibility for a bug to creep-in one of the two.

Reply:
also, grabbing the resourceVersion before/after and ensuring it didn't change would be a more reliable way of making sure nothing was persisted (assuming nothing else is touching pods)